org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/codeactions/SemicolonRequiredInReferenceCodeAction.java

```diff
@@ -0,0 +1,43 @@
+/**
+ *  Copyright (c) 2018 Angelo ZERR.
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v2.0
+ *  which accompanies this distribution, and is available at
+ *  http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ *  Contributors:
+ *  Angelo Zerr <angelo.zerr@gmail.com> - initial API and implementation
+ */
+package org.eclipse.lemminx.extensions.contentmodel.participants.codeactions;
+
+import java.util.List;
+
+import org.eclipse.lemminx.commons.CodeActionFactory;
+import org.eclipse.lemminx.dom.DOMDocument;
+import org.eclipse.lemminx.services.extensions.ICodeActionParticipant;
+import org.eclipse.lemminx.services.extensions.IComponentProvider;
+import org.eclipse.lemminx.settings.SharedSettings;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.Range;
+
+/**
+ * Code action to fix SemicolonRequiredInReference error.
+ *
```

Comment:
Add a sample in Javadoc.

Reply:
I mean given this XML:

```
<root>
  &mdash
</root>
```

The code action will fix by adding ';':

<root>
  &mdash;
</root>

Reply:
Indeed the main idea is to see quickly what the code action does. Our current code actions have no this sample and it is hard to understand the behavior of the code action. But let's start with your code action to provide good practice and we will then other code actions with sample.

Reply:
Sounds good! I can also take some time to update the comments for previous code actions if needed.